src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -58544,6 +58544,131 @@ void Test1(Test? x1)
+    {
+        E1 -= e2;
+        E1.Invoke(); // 2
+    }
+
+    " + modifiers + @"void M6()
+    {
+        E1 -= null;
+        E1.Invoke();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (27,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(27, 9),
+                // (33,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(33, 9)
+                );
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("static ")]
+        [WorkItem(42557, "https://github.com/dotnet/roslyn/issues/42557")]
+        public void EventAccessor_NullableEvent(string modifiers)
+        {
+            var source = @"
+class C
+{
+    " + modifiers + @"event System.Action? E1;
+
+    " + modifiers + @"void M1()
+    {
+        E1 += () => { };
+        E1.Invoke();
+    }
+
+    " + modifiers + @"void M2(System.Action? e2)
+    {
+        E1 += e2;
+        E1.Invoke(); // 1
+    }
+
+    " + modifiers + @"void M3()
+    {
+        E1 += null;
+        E1.Invoke(); // 2
+    }
+
+    " + modifiers + @"void M4()
+    {
+        E1 -= () => { };
+        E1.Invoke(); // 3
+    }
+
+    " + modifiers + @"void M(System.Action? e2)
+    {
+        E1 -= e2;
+        E1.Invoke(); // 4
+    }
+
+    " + modifiers + @"void M()
+    {
+        E1 -= null;
+        E1.Invoke(); // 5
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (15,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(15, 9),
+                // (21,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(21, 9),
+                // (27,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(27, 9),
+                // (33,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(33, 9),
+                // (39,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(39, 9)
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

Please test adding a delegate to an instance event in a static method.

Reply:
done in `InvalidEventAssignment_01`